File: python/1040.py
#1040 - Média 3

grades = input().split()

n1 = float(grades[0])
n2 = float(grades[1])
n3 = float(grades[2])
n4 = float(grades[3])

peso_n1 = 2
peso_n2 = 3
peso_n3 = 4
peso_n4 = 1

soma_pesos = peso_n1 + peso_n2 + peso_n3 + peso_n4

media = (n1 * peso_n1 + n2 * peso_n2 + n3 * peso_n3 + n4 * peso_n4) / soma_pesos

print(f"Media: {media:.1f}")

if media >= 7:
    print("Aluno aprovado.")
    
elif media < 5:
    print("Aluno reprovado.")

elif media >= 5 and media <= 6.9:
    print("Aluno em exame.")
    exam = float(input())
    print(f"Nota do exame: {exam:.1f}")
    media = (media + exam) / 2
    
    if media > 5:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    print(f"Media final: {media:.1f}")

File: iniciante/c99/1040/1040.c
#include <stdio.h>

int main() {

	double n1, n2, n3, n4, media;

	scanf("%lf", &n1);
	scanf("%lf", &n2);
	scanf("%lf", &n3);
	scanf("%lf", &n4);

	media = (n1 * 2 + n2 * 3 + n3 * 4 + n4 * 1) / 10;

	printf("Media: %.1lf\n", media);

	if (media >= 7)
		printf("Aluno aprovado.\n");

	if (media >= 5 && media < 7) {
			printf("Aluno em exame.\n");
				
			double nota_exame;
			scanf("%lf", &nota_exame);
			
			printf("Nota do exame: %.1lf\n", nota_exame);

			double media_final = (media + nota_exame) / 2; 

			if (media_final >= 5) 
				printf("Aluno aprovado.\n");

			else
				media = 0;
						
			printf("Media final: %.1lf\n", media_final);

	} 

	if (media < 5) 
		printf("Aluno reprovado.\n");

	return 0;

}
